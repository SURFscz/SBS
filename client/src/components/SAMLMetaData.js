import React from "react";
import "./SAMLMetaData.scss";
import I18n from "../locale/I18n";
import {isEmpty} from "../utils/Utils";

export const SAMLMetaData = ({parsedSAMLMetaData}) => {

    return (
        <div className="parsed-saml-meta-data first-column">
            <table>
                <thead/>
                <tbody>
                <tr>
                    <td className="attribute">{I18n.t("service.samlMetaData.acs_binding")}</td>
                    <td className="value">{parsedSAMLMetaData.acs_binding || I18n.t("service.samlMetaData.unknown")}</td>
                </tr>
                <tr>
                    <td className="attribute">{I18n.t("service.samlMetaData.acs_location")}</td>
                    <td className="value">{parsedSAMLMetaData.acs_location || I18n.t("service.samlMetaData.unknown")}</td>
                </tr>
                <tr>
                    <td className="attribute">{I18n.t("service.samlMetaData.entity_id")}</td>
                    <td className="value">{parsedSAMLMetaData.entity_id || I18n.t("service.samlMetaData.unknown")}</td>
                </tr>
                {!isEmpty(parsedSAMLMetaData.organization_name) && <tr>
                    <td className="attribute">{I18n.t("service.samlMetaData.organization_name")}</td>
                    <td className="value">{parsedSAMLMetaData.organization_name}</td>
                </tr>}
                </tbody>
            </table>
        </div>)
}