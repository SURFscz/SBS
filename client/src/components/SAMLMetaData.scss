@import "../stylesheets/vars.scss";

.parsed-saml-meta-data {
    margin: 15px 0;
    background-color: $hover;
    border: 1px solid $dark-grey;
    padding: 6px 12px;
    border-radius: $br;
    font-size: 14px;
    font-family: "Courier New", serif;

    table {
        td {
            padding: 5px;
            vertical-align: top;
            word-break: break-word;

            &.attribute {
                width: 25%;
            }

            &.value {
                width: 75%;
            }
        }
    }
}
