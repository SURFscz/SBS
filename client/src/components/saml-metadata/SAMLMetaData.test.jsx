import React from "react";
import {render} from "@testing-library/react";
import {SAMLMetaData} from "./SAMLMetaData";

jest.mock("../../locale/I18n", () => ({
    t: key => key
}));

const mockParsedSAMLMetaData = {
    acs_binding: "POST",
    acs_location: "https://example.com/acs",
    entity_id: "entity123",
    organization_name: "Example Org"
};

describe("SAMLMetaData", () => {
    it("matches snapshot with default props", () => {
        const {asFragment} = render(
            <SAMLMetaData
                parsedSAMLMetaData={mockParsedSAMLMetaData}
                metadataUrl="https://example.com/saml/metadata"
                onMetadataChange={() => {
                }}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom metadataUrl", () => {
        const {asFragment} = render(
            <SAMLMetaData
                parsedSAMLMetaData={mockParsedSAMLMetaData}
                metadataUrl="https://custom.com/saml/metadata"
                onMetadataChange={() => {
                }}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with no metadataUrl", () => {
        const {asFragment} = render(
            <SAMLMetaData
                parsedSAMLMetaData={mockParsedSAMLMetaData}
                onMetadataChange={() => {
                }}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
