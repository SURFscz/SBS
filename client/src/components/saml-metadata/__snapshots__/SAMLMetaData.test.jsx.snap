// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SAMLMetaData matches snapshot with custom metadataUrl 1`] = `
<DocumentFragment>
  <div
    class="parsed-saml-meta-data first-column"
  >
    <table>
      <thead />
      <tbody>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.acs_binding
          </td>
          <td
            class="value"
          >
            POST
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.acs_location
          </td>
          <td
            class="value"
          >
            https://example.com/acs
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.entity_id
          </td>
          <td
            class="value"
          >
            entity123
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.organization_name
          </td>
          <td
            class="value"
          >
            Example Org
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`SAMLMetaData matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="parsed-saml-meta-data first-column"
  >
    <table>
      <thead />
      <tbody>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.acs_binding
          </td>
          <td
            class="value"
          >
            POST
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.acs_location
          </td>
          <td
            class="value"
          >
            https://example.com/acs
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.entity_id
          </td>
          <td
            class="value"
          >
            entity123
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.organization_name
          </td>
          <td
            class="value"
          >
            Example Org
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`SAMLMetaData matches snapshot with no metadataUrl 1`] = `
<DocumentFragment>
  <div
    class="parsed-saml-meta-data first-column"
  >
    <table>
      <thead />
      <tbody>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.acs_binding
          </td>
          <td
            class="value"
          >
            POST
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.acs_location
          </td>
          <td
            class="value"
          >
            https://example.com/acs
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.entity_id
          </td>
          <td
            class="value"
          >
            entity123
          </td>
        </tr>
        <tr>
          <td
            class="attribute"
          >
            service.samlMetaData.organization_name
          </td>
          <td
            class="value"
          >
            Example Org
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
